### PR TITLE
7903636: update jextract sample scripts for --source option removal

### PR DESCRIPTION
**Pull request by sundararajana**
* removing --source option in scripts
* removing binary .class generating scripts (as normal scripts do compilation by javac anyway)
* removing unnecessary --enable-preview option in run scripts.
* updating gradle version to be the latest (8.5) so that we can use JDK 21 for gradle run itself.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed (no review required)

### Issue
 * [CODETOOLS-7903636](https://bugs.openjdk.org/browse/CODETOOLS-7903636): update jextract sample scripts for --source option removal (**Bug** - P3)

### Reviewers
 * [Maurizio Cimadamore](https://openjdk.org/census#mcimadamore) (@mcimadamore - **Reviewer**) ⚠️ Review applies to [37f0d648](https://git.openjdk.org/jextract/pull/189/files/37f0d648fbeb5b387137688916be071fba2a23d3)

### Contributors
 * Jorn Vernee `<jvernee@openjdk.org>`

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jextract.git pull/189/head:pull/189` \
`$ git checkout pull/189`

Update a local copy of the PR: \
`$ git checkout pull/189` \
`$ git pull https://git.openjdk.org/jextract.git pull/189/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 189`

View PR using the GUI difftool: \
`$ git pr show -t 189`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jextract/pull/189.diff">https://git.openjdk.org/jextract/pull/189.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jextract/pull/189#issuecomment-1903355315)

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.3.3-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.5-bin.zip
```
Comment (mcimadamore) [inline gradle/wrapper/gradle-wrapper.properties:3]: I don't think these changes should be included?

File: samples/cblas/run.sh [hunk 2]
```diff
@@ -1,3 +1,3 @@
-time java --enable-native-access=ALL-UNNAMED --enable-preview --source=22 \
+time java --enable-native-access=ALL-UNNAMED \
```
Comment (mcimadamore) [inline samples/cblas/run.sh:1]: why the `time` in front?

**Review by mcimadamore — COMMENTED** [review]
Looks good, simpler. Hopefully when some issues with source code launcher are fixed, we can simplify even more :-)

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (sundararajana) [reply gradle/wrapper/gradle-wrapper.properties:3]: With that change, we can use JDK 21 to build jextract. Without that, we've to use JDK 18. Do we want this or not?

File: samples/cblas/run.sh (hunk 2 above)
Comment (sundararajana) [reply samples/cblas/run.sh:1]: That was pre-existing code. Most likely we had it to have simple measure of execution speed of numeric libraries in the past. I can remove it if desired.

**Review by mcimadamore — APPROVED** [review]

**Pull request closed without merging**